libraries/AP_HAL_ChibiOS/hwdef/iomcu/hwdef.dat

```diff
@@ -117,6 +117,15 @@ define HAL_NO_MONITOR_THREAD
 define HAL_NO_RCOUT_THREAD
 define AP_HAL_SHARED_DMA_ENABLED 0
 
+# chibios tuning to save memory
+define CH_CFG_USE_REGISTRY FALSE
+define CH_CFG_USE_WAITEXIT FALSE
+define CH_DBG_FILL_THREADS FALSE
+define CH_CFG_OPTIMIZE_SPEED FALSE
+define CH_CFG_USE_DYNAMIC FALSE
+define CH_CFG_USE_MUTEXES_RECURSIVE FALSE
```

Comment:
this is dangerous if a dependency on recursive semaphores in a library that IOMCU uses comes in the future then we could deadlock